R/utils-numeric.R

```diff
@@ -56,8 +62,8 @@ getDiscretizedBins <- function(x, method = c('equalInterval', 'quantile', 'sd'),
     values <- rep(NA_real_, length(binStarts))
   }
   
-  bins <- lapply(1:length(binStarts), FUN = function(x) { Bin(binStart = as.character(binStarts[[x]]),
-                                                              binEnd = as.character(binEnds[[x]]),
+  bins <- lapply(1:length(binStarts), FUN = function(x) { Bin(binStart = binStarts[[x]],
+                                                              binEnd = binEnds[[x]],
```

Comment:
i dont know why i made this character... are we confident that numerics havent lost precision/ accuracy (differences in as.character and print or something maybe??) the bin start and ends must represent exactly the value used to create the bins, so that the bins can be recreated in other contexts as needed

Reply:
oh yes good catch. I made a quick decision about how to handle this for now that just differentiates between dates and everything else. It's not the most elegant but we can improve it later! Feeling very move-fast-and-break-things today